### ISSUE TITLE
[BUG]: error message should show more details

### ISSUE THREAD

**Issue by celestinoxp** [pycaret/pycaret#2829]
### pycaret version checks

- [X] I have checked that this issue has not already been reported [here](https://github.com/pycaret/pycaret/issues).

- [X] I have confirmed this bug exists on the [latest version](https://github.com/pycaret/pycaret/releases) of pycaret.

- [X] I have confirmed this bug exists on the master branch of pycaret (pip install -U git+https://github.com/pycaret/pycaret.git@master).

### Issue Description

when we receive this message: 
ValueError: The levels passed to the ordinal_features parameter doesn't match with the levels in the dataset

we don't know which feature is causing problems.

Test Case
So, imagine we have a dataset of football (or soccer) , and in the entire dataset a visiting team had only one time, 2 red cards in that game, that is, 1 unique event in the entire dataset. Now pycaret gives an error because a variable has a value that only happened once

Problem i receive a error message and i don´t know were is the problem.

My solution:  I opened the file in question and edited the code. I inserted this line: f"{self.X[key]}", it works because show me the feature in cause, but i think you can give a better solution :)

### Reproducible Example

```python
s = setup(data=dados_treino, target='vencedor_tr', numeric_imputation='knn', ignore_features=['ronda', 'tipo_de_jogo'], fix_imbalance=True, normalize=True, max_encoding_ohe=8, fold=5, use_gpu=True, session_id=1)
```

### Expected Behavior

possible error but showing variable in cause

### Actual Results

```python-traceback
ValueError                                Traceback (most recent call last)
Input In [20], in <cell line: 1>()
----> 1 s = setup(data=dados_treino, target='vencedor_tr', numeric_imputation='knn', ignore_features=['ronda', 'tipo_de_jogo'], fix_imbalance=True, normalize=True, max_encoding_ohe=8, fold=5, use_gpu=True, session_id=1)

File ~\anaconda3\envs\teste2\lib\site-packages\pycaret\classification\functional.py:590, in setup(data, data_func, target, train_size, test_data, ordinal_features, numeric_features, categorical_features, date_features, text_features, ignore_features, keep_features, preprocess, create_date_columns, imputation_type, numeric_imputation, categorical_imputation, iterative_imputation_iters, numeric_iterative_imputer, categorical_iterative_imputer, text_features_method, max_encoding_ohe, encoding_method, rare_to_value, rare_value, polynomial_features, polynomial_degree, low_variance_threshold, group_features, group_names, remove_multicollinearity, multicollinearity_threshold, bin_numeric_features, remove_outliers, outliers_method, outliers_threshold, fix_imbalance, fix_imbalance_method, transformation, transformation_method, normalize, normalize_method, pca, pca_method, pca_components, feature_selection, feature_selection_method, feature_selection_estimator, n_features_to_select, custom_pipeline, custom_pipeline_position, data_split_shuffle, data_split_stratify, fold_strategy, fold, fold_shuffle, fold_groups, n_jobs, use_gpu, html, session_id, system_log, log_experiment, experiment_name, experiment_custom_tags, log_plots, log_profile, log_data, verbose, memory, profile, profile_kwargs)
    588 exp = _EXPERIMENT_CLASS()
    589 set_current_experiment(exp)
--> 590 return exp.setup(
    591     data=data,
    592     data_func=data_func,
    593     target=target,
    594     train_size=train_size,
    595     test_data=test_data,
    596     ordinal_features=ordinal_features,
    597     numeric_features=numeric_features,
    598     categorical_features=categorical_features,
    599     date_features=date_features,
    600     text_features=text_features,
    601     ignore_features=ignore_features,
    602     keep_features=keep_features,
    603     preprocess=preprocess,
    604     create_date_columns=create_date_columns,
    605     imputation_type=imputation_type,
    606     numeric_imputation=numeric_imputation,
    607     categorical_imputation=categorical_imputation,
    608     iterative_imputation_iters=iterative_imputation_iters,
    609     numeric_iterative_imputer=numeric_iterative_imputer,
    610     categorical_iterative_imputer=categorical_iterative_imputer,
    611     text_features_method=text_features_method,
    612     max_encoding_ohe=max_encoding_ohe,
    613     encoding_method=encoding_method,
    614     rare_to_value=rare_to_value,
    615     rare_value=rare_value,
    616     polynomial_features=polynomial_features,
    617     polynomial_degree=polynomial_degree,
    618     low_variance_threshold=low_variance_threshold,
    619     group_features=group_features,
    620     group_names=group_names,
    621     remove_multicollinearity=remove_multicollinearity,
    622     multicollinearity_threshold=multicollinearity_threshold,
    623     bin_numeric_features=bin_numeric_features,
    624     remove_outliers=remove_outliers,
    625     outliers_method=outliers_method,
    626     outliers_threshold=outliers_threshold,
    627     fix_imbalance=fix_imbalance,
    628     fix_imbalance_method=fix_imbalance_method,
    629     transformation=transformation,
    630     transformation_method=transformation_method,
    631     normalize=normalize,
    632     normalize_method=normalize_method,
    633     pca=pca,
    634     pca_method=pca_method,
    635     pca_components=pca_components,
    636     feature_selection=feature_selection,
    637     feature_selection_method=feature_selection_method,
    638     feature_selection_estimator=feature_selection_estimator,
    639     n_features_to_select=n_features_to_select,
    640     custom_pipeline=custom_pipeline,
    641     custom_pipeline_position=custom_pipeline_position,
    642     data_split_shuffle=data_split_shuffle,
    643     data_split_stratify=data_split_stratify,
    644     fold_strategy=fold_strategy,
    645     fold=fold,
    646     fold_shuffle=fold_shuffle,
    647     fold_groups=fold_groups,
    648     n_jobs=n_jobs,
    649     use_gpu=use_gpu,
    650     html=html,
    651     session_id=session_id,
    652     system_log=system_log,
    653     log_experiment=log_experiment,
    654     experiment_name=experiment_name,
    655     experiment_custom_tags=experiment_custom_tags,
    656     log_plots=log_plots,
    657     log_profile=log_profile,
    658     log_data=log_data,
    659     verbose=verbose,
    660     memory=memory,
    661     profile=profile,
    662     profile_kwargs=profile_kwargs,
    663 )

File ~\anaconda3\envs\teste2\lib\site-packages\pycaret\classification\oop.py:818, in ClassificationExperiment.setup(self, data, data_func, target, train_size, test_data, ordinal_features, numeric_features, categorical_features, date_features, text_features, ignore_features, keep_features, preprocess, create_date_columns, imputation_type, numeric_imputation, categorical_imputation, iterative_imputation_iters, numeric_iterative_imputer, categorical_iterative_imputer, text_features_method, max_encoding_ohe, encoding_method, rare_to_value, rare_value, polynomial_features, polynomial_degree, low_variance_threshold, group_features, group_names, remove_multicollinearity, multicollinearity_threshold, bin_numeric_features, remove_outliers, outliers_method, outliers_threshold, fix_imbalance, fix_imbalance_method, transformation, transformation_method, normalize, normalize_method, pca, pca_method, pca_components, feature_selection, feature_selection_method, feature_selection_estimator, n_features_to_select, custom_pipeline, custom_pipeline_position, data_split_shuffle, data_split_stratify, fold_strategy, fold, fold_shuffle, fold_groups, n_jobs, use_gpu, html, session_id, system_log, log_experiment, experiment_name, experiment_custom_tags, log_plots, log_profile, log_data, engines, verbose, memory, profile, profile_kwargs)
    816 # Encode non-numerical features
    817 if self._fxs["Ordinal"] or self._fxs["Categorical"]:
--> 818     self._encoding(
    819         max_encoding_ohe=max_encoding_ohe,
    820         encoding_method=encoding_method,
    821         rare_to_value=rare_to_value,
    822         rare_value=rare_value,
    823     )
    825 # Create polynomial features from the existing ones
    826 if polynomial_features:

File ~\anaconda3\envs\teste2\lib\site-packages\pycaret\internal\preprocess\preprocessor.py:556, in Preprocessor._encoding(self, max_encoding_ohe, encoding_method, rare_to_value, rare_value)
    554 for key, value in self._fxs["Ordinal"].items():
    555     if self.X[key].nunique() != len(value):
--> 556         raise ValueError(
    557             f"The levels passed to the ordinal_features parameter "
    558             f"doesn't match with the levels in the dataset."
    559             f"{self.X[key]}"
    560         )
    562     # Encoder always needs mapping of NaN value
    563     mapping[key] = {v: i for i, v in enumerate(value)}

ValueError: The levels passed to the ordinal_features parameter doesn't match with the levels in the dataset.0      0
1      0
2      0
3      0
4      0
      ..
148    0
149    0
150    0
151    0
152    0
Name: cartoes_vermelhos_fora, Length: 153, dtype: int8
```

### Installed Versions

<details>
System:
    python: 3.9.12 (main, Apr  4 2022, 05:22:27) [MSC v.1916 64 bit (AMD64)]
executable: C:\Users\username\anaconda3\envs\teste2\python.exe
   machine: Windows-10-10.0.22622-SP0

PyCaret required dependencies:
                 pip: 22.2.1
          setuptools: 61.2.0
             pycaret: 3.0.0rc3
             IPython: 8.4.0
          ipywidgets: 7.7.1
                tqdm: 4.64.0
               numpy: 1.22.4
              pandas: 1.4.3
              jinja2: 3.1.2
               scipy: 1.8.1
              joblib: 1.1.0
             sklearn: 1.1.1
                pyod: 1.0.3
            imblearn: 0.9.1
   category_encoders: 2.5.0
            lightgbm: 3.3.2.99
               numba: 0.55.2
            requests: 2.28.1
          matplotlib: 3.5.2
          scikitplot: 0.3.7
         yellowbrick: 1.4
              plotly: 5.9.0
             kaleido: 0.2.1
         statsmodels: 0.13.2
              sktime: 0.13.0
               tbats: 1.1.0
            pmdarima: 1.8.5
              psutil: 5.9.1

PyCaret optional dependencies:
                shap: 0.40.0
           interpret: 0.2.7
                umap: 0.5.3
    pandas_profiling: 3.2.0
  explainerdashboard: 0.4.0
             autoviz: 0.1.42
           fairlearn: 0.7.0
             xgboost: 2.0.0.dev0
            catboost: 1.0.6
              kmodes: 0.12.1
             mlxtend: 0.20.0
       statsforecast: 0.7.0
        tune_sklearn: 0.4.3
                 ray: 1.13.0
            hyperopt: 0.2.7
              optuna: 2.10.1
               skopt: 0.9.0
              mlflow: 1.27.1.dev0
              gradio: 3.1.1
             fastapi: 0.79.0
             uvicorn: 0.18.2
              m2cgen: 0.10.0
           evidently: 0.1.53.dev0
                nltk: 3.7
            pyLDAvis: Not installed
              gensim: Not installed
               spacy: 3.4.0
           wordcloud: 1.8.2.2
            textblob: 0.17.1
               fugue: 0.6.6
           streamlit: Not installed
             prophet: 1.1
</details>

**Comment by celestinoxp** [comment]
and we also have to solve this problem of having a feature with only 1 occurrence and pycaret gives an error, because this feature is not the target.
Imagine it happened only once in a lifetime for a football team to have 2 red cards, this being a visiting team. But my target is the number of goals, that is, it has nothing to do with red cards, but they are very important, so pycaret cannot force me to ignore this feature.

**Comment by celestinoxp** [comment]
Can check line 559? i added manualy in my notebook  this line f"{self.X[key]}", but i think you should analyse that.

@tvdboom are you the author of this file ... pycaret\internal\preprocess\preprocessor.py? Can you check that?

**Comment by tvdboom** [comment]
Hi, the issue is fixed in #2839. I clarified the message and changed it from error to warning. The reason it should not be an error is because users can know of classes that will appear with new data but are not present during fit.

**Comment by tvdboom** [comment]
> and we also have to solve this problem of having a feature with only 1 occurrence and pycaret gives an error, because this feature is not the target.

Can you open a new issue with a reproducible example? I don't know which error pycaret is producing.

**Comment by celestinoxp** [comment]
> Olá, o problema foi corrigido em #2839 . Esclareci a mensagem e mudei de erro para aviso. A razão pela qual não deve ser um erro é porque os usuários podem saber de classes que aparecerão com novos dados, mas não estarão presentes durante o ajuste.

Thank you! :)

**Comment by celestinoxp** [comment]
> > and we also have to solve this problem of having a feature with only 1 occurrence and pycaret gives an error, because this feature is not the target.
> 
> Can you open a new issue with a reproducible example? I don't know which error pycaret is producing.

yea. as soon as i can i will create a new issue
